codec/src/types/primitives.rs

```diff
@@ -40,6 +43,39 @@ impl_numeric!(i128, get_i128, put_i128);
 impl_numeric!(f32, get_f32, put_f32);
 impl_numeric!(f64, get_f64, put_f64);
 
+// Usize implementation
+// Since most values refer to a length or size, the type is encoded as a varint.
+// We force the maximum size to be 4 bytes for consistency across platforms. In addition, we require
```

Comment:
Might be worth clarifying that this panics if the encoding size would be more than 4 bytes. "Force" makes it sound like it caps it or something?